src/assets/wise5/components/aiChat/ai-chat-student/ai-chat-student.component.ts

```diff
@@ -51,12 +58,96 @@ export class AiChatStudentComponent extends ComponentStudent {
 
   ngOnInit(): void {
     super.ngOnInit();
+    if (this.component.isComputerAvatarEnabled()) {
+      this.initializeComputerAvatar();
+    } else {
+      this.computerAvatar = this.computerAvatarService.getDefaultAvatar();
+    }
     if (this.componentState != null) {
       this.messages = this.componentState.studentData.messages;
     }
     this.initializeMessages();
   }
 
+  initializeComputerAvatar(): void {
+    this.tryToRepopulateComputerAvatar();
+    if (this.hasStudentPreviouslyChosenComputerAvatar()) {
+      this.hideComputerAvatarSelector();
+    } else if (
+      this.component.isOnlyOneComputerAvatarAvailable() &&
+      !this.component.isComputerAvatarPromptAvailable()
+    ) {
+      this.hideComputerAvatarSelector();
+      this.selectComputerAvatar(this.getTheOnlyComputerAvatarAvailable());
+    } else {
+      this.showComputerAvatarSelector();
+    }
+  }
+
+  private tryToRepopulateComputerAvatar(): void {
+    if (this.includesComputerAvatar(this.componentState)) {
+      this.repopulateComputerAvatarFromComponentState(this.componentState);
+    } else if (
+      this.component.isUseGlobalComputerAvatar() &&
+      this.isGlobalComputerAvatarAvailable()
```

Comment:
Should this be a function in the StudentStatusService instead? This way, we don't need to define the function in AIChatStudent and DialogGuidanceStudent
```suggestion
      this.studentStatusService.isGlobalComputerAvatarAvailable()
```